cmd/util/ledger/migrations/storage_reporter.go

```diff
@@ -35,76 +65,96 @@ func (r StorageReporter) Report(payload []ledger.Payload) error {
 		if id.Key != "storage_used" {
 			continue
 		}
+		address := flow.BytesToAddress([]byte(id.Owner))
 		u, _, err := utils.ReadUint64(p.Value)
 		if err != nil {
 			return err
 		}
-		storageUsed[id.Owner] = u
-		average = average + (float64(u)-average)/(float64(i)+1.0)
-		if u > max {
-			max = u
-		}
-	}
-	r.Log.Info().
-		Msgf("Average storage used %g", average)
-	r.Log.Info().
-		Msgf("Max storage used %d", max)
-
-	bins := 150
-	binHeight := 30
-	w := float64(max) / float64(bins)
-	distribution := make([]float64, bins)
-
-	var maxBin = float64(0)
-
-	for s, u := range storageUsed {
-		var i = int(float64(u) / w)
-		if i >= bins {
-			i = bins - 1
+		balance, hasVault, err := r.balance(address, st)
+		if err != nil {
+			r.Log.Err(err).Msg("Cannot get account balance")
+			return err
 		}
-		distribution[i] = distribution[i] + 1
-
-		if maxBin < distribution[i] {
-			maxBin = distribution[i]
+		dapper, err := r.isDapper(address, st)
+		if err != nil {
+			r.Log.Err(err).Msg("Cannot determine if this is a dapper account")
+			return err
 		}
-		if i == bins-1 {
-			r.Log.Info().
-				Msgf("High storage usage address: %s, used: %d bytes", flow.BytesToAddress([]byte(s)), u)
+		record := fmt.Sprintf("%s,%d,%d,%t,%t\n", address.Hex(), u, balance, hasVault, dapper)
+		_, err = writer.WriteString(record)
+		if err != nil {
+			return err
 		}
 	}
-	maxBinAccounts := maxBin
-	maxBin = math.Log10(maxBin)
 
-	graph := make([]int, bins)
+	r.Log.Info().Msg("Storage Reporter Done.")
 
-	for i, d := range distribution {
-		var v float64
-		if d <= 0 {
-			v = 0
-		} else {
-			v = math.Log10(d)
-		}
+	return nil
+}
 
-		graph[i] = int(math.Ceil((v / maxBin) * float64(binHeight)))
+func (r StorageReporter) isDapper(address flow.Address, st *state.State) (bool, error) {
+	id := flow.RegisterID{
+		Owner:      string(address.Bytes()),
+		Controller: "",
+		Key:        fmt.Sprintf("%s\x1F%s", "public", "dapperUtilityCoinReceiver"),
```

Comment:
Would be nice if we could use a function here. Maybe refactor this and the other instances below into a function and link to the Cadence function that does this.